### ISSUE TITLE
Is it possible to specify an alias for a nested joinrelation?

### ISSUE THREAD

**Issue by lukecwilliams** [Vincit/objection.js#545]
Example: in this code I'd like to alias workoutCategory.  So instead of:

    .joinRelation('standardWorkoutDefinition.workoutCategory')
    .where('standardWorkoutDefinition.workoutCategory.id', workoutCategoryId)

I would have:

    .joinRelation('standardWorkoutDefinition.workoutCategory as cat')
    .where('cat.id', workoutCategoryId)

Using .alias() actually aliases only standardWorkoutDefinition.  Is there a way to alias workoutCategory in this example?

**Comment by koskimas** [comment]
Its impossible to alias nested relations like `standardWorkoutDefinition.workoutCategory` into a single alias like `cat`. The relation expression language supports aliases, but only for single relations:

```js
Person
  .query()
  .joinEager(`[
    children as kids.[
      pets(filterDogs) as dogs,
      pets(filterCats) as cats
    ]
  ]`)
  .where('kids:cats.id', 1)
```

If that doesn't already work with `joinRelation`, it probably should. Could you try if it works? Could you live with giving two aliases like that?

**Comment by koskimas** [comment]
I tested and it didn't work. I'll fix it right now.

**Comment by koskimas** [comment]
Done.

**Comment by lukecwilliams** [comment]
Wow - awesome.  Thanks for such fast follow-up.  

We're big Objection.js fans and actively evangelize it.  Even bigger fans now.

**Comment by koskimas** [comment]
No problem at all! I'm just happy when someone finds a bug. I hate bugs :D

The fix will be released with 0.9, and that can take a while. I keep all the test passing in the master branch which means it's usually pretty stable. You can try (or even temporarily use) the fix by pointing your package.json to a specific github commit.
